### PR TITLE
Fix: 1008 `argo wait` and `argo submit --wait` should exit 1 if workflow fails 

### LINKED ISSUE #1008: `argo wait` and `argo submit --wait` should exit 1 if workflow fails
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG REPORT

**What happened**:

When argo wait completes and the workflow failed, it should return non-zero.

**Environment**:
- Argo version: v2.2.0

### PR DESCRIPTION
**Pull request by sarabala1979**
This PR will resolve #1008

### CODE REVIEW

**Comment by xianlubird** [comment]
LGTM

File: cmd/argo/commands/wait.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ func waitOnOne(workflowName string, ignoreNotFound, quiet bool) {
 			if !quiet {
 				fmt.Printf("%s completed at %v\n", workflowName, wf.Status.FinishedAt)
 			}
+			if wf.Status.Phase == wfv1.NodeFailed || wf.Status.Phase == wfv1.NodeError {
+				os.Exit(1)
```
Comment (jessesuen) [inline cmd/argo/commands/wait.go]: This behavior makes it fail fast. Lets return error and exit(1) after all workflows complete.

File: cmd/argo/commands/wait.go [hunk 2]
```diff
@@ -36,17 +36,25 @@ func NewWaitCommand() *cobra.Command {
 // WaitWorkflows waits for the given workflowNames.
 func WaitWorkflows(workflowNames []string, ignoreNotFound, quiet bool) {
 	var wg sync.WaitGroup
+	wfSuccessStatus := make([]bool, 0, len(workflowNames))
```
Comment (jessesuen) [inline cmd/argo/commands/wait.go]: Minor: why not use a bool initialized to true, and set to false as soon as you see one that failed? Then you don't have to iterate this list, and simply exit 1 if the bool is false.

**Comment by rpocase** [comment]
Does this fix apply to `watch` as well? When running locally we wrap argo in `--watch` for a better dev experience. This does not seem to be exiting non-zero when the workflow fails. In CI we use `--wait`.

**Pull request merged**